### PR TITLE
Java and documentation fix

### PR DESCRIPTION
**Pull request by ferazoguerrero**
## Ticket
[CAS-82](https://twilio-engineering.atlassian.net/browse/CAS-82)

## Description
- Solve issue "java not found" when running the release pipeline.
- Add a pipeline to invoke dokka and generates documentation.

### CODE REVIEW

File: .circleci/release_kmp.yml [hunk 1]
```diff
@@ -281,6 +283,21 @@ jobs:
             TMP_WORKSPACE: *tmp-workspace
             ENV_VARS_FILE: *env-vars-file
 
+  generate-docs:
```
Comment (SergioFierro) [inline .circleci/release_kmp.yml:286]: The artifact generation is also running the dokka task to generate the docs https://github.com/twilio/twilio-verify-passkeys/blob/dev/shared/build.gradle.kts#L71. Is this job still necessary?

File: .circleci/config.yml [hunk 2]
```diff
@@ -634,7 +634,3 @@ workflows:
       - should-release-kmp-version:
           requires:
             - check-android-local-sdk-e2e-tests-result
-          filters:
-            branches:
-              only:
-                - main
```
Comment (SergioFierro) [inline .circleci/config.yml:640]: why was this removed?

File: .circleci/release_kmp.yml (hunk 1 above)
Comment (ferazoguerrero) [reply .circleci/release_kmp.yml:286]: seems to me dokka task is not invoked. [Log](https://app.circleci.com/pipelines/github/twilio/twilio-verify-passkeys/426/workflows/73323489-83fc-4282-aca1-0ca147c8bfc4/jobs/4612)

File: .circleci/config.yml (hunk 2 above)
Comment (ferazoguerrero) [reply .circleci/config.yml:640]: just for testing release pipelines in this branch. But the filter must be restored

**Review by SergioFierro — APPROVED** [review]
